src/gui/shellwidget/shellwidget.cpp

```diff
@@ -273,8 +266,13 @@ void ShellWidget::paintUndercurl(
 		return;
 	}
 
-	p.setPen(getSpecialPen(cell));
-	p.drawPath(GetUndercurlPath(cellRect));
+	QPen specialPen = getSpecialPen(cell);
+	p.setPen(specialPen);
+	// only enable anti-aliasing for the undercurl (causes rendering
+	// artifacts when rendering the cell background)
+	p.setRenderHint(QPainter::Antialiasing);
```

Comment:
Since `p` is an out-param it has the side-effect of setting `QPainter::Antialiasing`.

Let's be extra cautious and set/un-set the value, since antialiasing causes issues elsewhere:
```
const auto renderHints { p.renderHints() };
... /* Your existing code */
p.setRenderHints(renderHints);
```

Do quickly verify this code actually works...

I didn't observe anything wrong with your code, just trying to prevent accidental bugs :)

Reply:
Seems like a good idea.  I've added it to the PR.